packages/components/src/Heading.tsx

```diff
@@ -0,0 +1,30 @@
+import React from 'react'
+import Box, { Assign, BoxOwnProps, ForwardRef } from './Box'
+
+export interface HeadingProps
+  extends Assign<React.ComponentPropsWithRef<'h2'>, BoxOwnProps> {}
+/**
+ * Primitive heading component, defaults to <h2>.
+ *
+ * Text style variants can be defined in the theme.text object.
+ * The Heading component uses theme.text.heading as its default variant style.
+ * @see https://theme-ui.com/components/heading
+ */
+
+export const Heading: ForwardRef<
+  HTMLHeadingElement,
+  HeadingProps
+> = React.forwardRef((props, ref) => (
+  <Box
+    ref={ref}
+    as="h2"
+    variant="heading"
+    {...props}
+    __themeKey="text"
+    css={{
+      fontFamily: 'heading',
+      fontWeight: 'heading', //TODO
```

Comment:
Isn't this a bug? `css` is passed to styled component without touching theme. Shouldn't this be `__css`? 

cc @jxnblk

https://github.com/system-ui/theme-ui/blob/master/packages/components/src/Box.js#L13-L30

Reply:
Hey @hasparus thanks for looking into this... what would you say is next step is? I'd like to help out but you lost me at the forward ref generics. 

Is there anything i can be investigating or is there a bigger conversations that needs to happen about this package?

Reply:
The ForwardRef generic could be probably dropped tbh. TS would infer proper types if we annotate component props and ref. The problem is, components with "as" prop are tricky to type. We should write some tests to check their public api on type level (inference of onClick event would be perfect, not sure if we can achieve it with Box).

We could change "as" prop usages to withComponent, to get rid of the errors, but I’m not sure if we want to, as this would have a small runtime cost. We’d probably be fine with few as any assertions.

Reply:
Good catch! This does look like a bug, e.g. `css` prop should be `__css`